### ISSUE TITLE
Create a GitHub Actions workflow to build and test the code in this repo

### ISSUE THREAD

**Issue by saffaffi** [gememma/wirebrush#4]
Any modern codebase should be automatically built and tested by a CI system, to make sure that any changes merged into the main branch are up to standard and don't introduce any regressions.

This repo should make use of a GitHub Actions workflow to do the following things:

- [x] Install a Rust toolchain, making sure it's using the correct version as specified by the `rust-toolchain.toml` file - see #1.
- [x] Build all the code with Cargo
- [x] Run all the tests with Cargo
- [ ] If you're doing this after #3, make sure all the commands run by the workflow apply to the whole workspace. Check whether the `cargo` commands it invokes need to have a `--workspace` or `--all` flag passed to them to tell them to perform their action on the whole workspace, or whether any jobs in the workflow need to have a build matrix added to repeat them for each of several crates. Note that even if a command doesn't _require_ a `--workspace`/`--all` flag, it might be nice to add it for peace of mind (use your judgement here).
